Bottom: eefa43641cda28f17197c794311125103b11ff0a
Top:    354dbb14fd2be465d26fc0e496bf084453d1560f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-01-21 21:26:23 +0000

hid/gtk: Remove XOR colour logic from the GL rederer.

This code was not being used, so remove it.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index f3133ec..45b9b72 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -68,7 +68,6 @@ typedef struct hid_gc_struct
   double alpha_mult;
   Coord width;
   gint cap, join;
-  gchar xor;
 }
 hid_gc_struct;
 
@@ -356,8 +355,6 @@ typedef struct
 {
   int color_set;
   GdkColor color;
-  int xor_set;
-  GdkColor xor_color;
   double red;
   double green;
   double blue;
@@ -420,20 +417,6 @@ set_gl_color_for_gc (hidGC gc)
           cc->blue  = cc->color.blue  / 65535.;
           cc->color_set = 1;
         }
-      if (gc->xor)
-        {
-          if (!cc->xor_set)
-            {
-              cc->xor_color.red = cc->color.red ^ gport->bg_color.red;
-              cc->xor_color.green = cc->color.green ^ gport->bg_color.green;
-              cc->xor_color.blue = cc->color.blue ^ gport->bg_color.blue;
-              gdk_color_alloc (gport->colormap, &cc->xor_color);
-              cc->red   = cc->color.red   / 65535.;
-              cc->green = cc->color.green / 65535.;
-              cc->blue  = cc->color.blue  / 65535.;
-              cc->xor_set = 1;
-            }
-        }
       r = cc->red;
       g = cc->green;
       b = cc->blue;
